client/my-sites/checkout/upsell-nudge/index.jsx

```diff
@@ -552,14 +555,17 @@ const trackUpsellButtonClick = ( eventName ) => {
 	return recordTracksEvent( eventName, { section: 'checkout' } );
 };
 
-const resolveProductSlug = ( upsellType, productAlias ) => {
+const resolveProductSlug = ( upsellType, productAlias, billingTerm ) => {
 	switch ( upsellType ) {
 		case BUSINESS_PLAN_UPGRADE_UPSELL:
 			return getPlanByPathSlug( productAlias )?.getStoreSlug();
 		case ANNUAL_PLAN_UPGRADE:
 			return productAlias;
 		case PROFESSIONAL_EMAIL_UPSELL:
-			return TITAN_MAIL_MONTHLY_SLUG;
+			if ( billingTerm === TERM_MONTHLY ) {
+				return TITAN_MAIL_MONTHLY_SLUG;
+			}
+			return TITAN_MAIL_YEARLY_SLUG;
```

Comment:
Not mandatory at all but since the resulting line wouldn't be that long we could also use that approach:

```suggestion
			return billingTerm === TERM_MONTHLY ? TITAN_MAIL_MONTHLY_SLUG : TITAN_MAIL_YEARLY_SLUG;
```

Reply:
We could at least add blank lines to this function to make it easier to read:

```javascript
const resolveProductSlug = ( upsellType, productAlias, billingTerm ) => {
  switch ( upsellType ) {
    case BUSINESS_PLAN_UPGRADE_UPSELL:
      return getPlanByPathSlug( productAlias )?.getStoreSlug();
      
    case ANNUAL_PLAN_UPGRADE:
      return productAlias;
      
    case PROFESSIONAL_EMAIL_UPSELL:
      if ( billingTerm === TERM_MONTHLY ) {
        return TITAN_MAIL_MONTHLY_SLUG;
      }
      
      return TITAN_MAIL_YEARLY_SLUG;
      
    case CONCIERGE_QUICKSTART_SESSION:
    case CONCIERGE_SUPPORT_SESSION:
    default:
      return 'concierge-session';
  }
};
```